File: InputTestCode/Swift/KotlinBasics/14_TypeChecksAutomaticCasts.swift
// Type checks and automatic casts

func getStringLength1(obj: Any) -> Int? {
    if let str = obj as? String {
        return str.count
    }
    
    return nil
}

func getStringLength2(obj: Any) -> Int? {
    if (!(obj is String)) {
        return nil
    }

    return (obj as! String).count
}

func getStringLength3(obj: Any) -> Int? {
    if let str = obj as? String {
        if str.count > 0 {
            return str.count
        }
    }

    return nil
}

func automaticCastOneVariable(arg1: Int?) -> Int? {
    if (arg1 != nil) {
        return arg1! * 2
    } else {
        return nil
    }
}

File: InputTestCode/Kotlin/KotlinBasics/14_TypeChecksAutomaticCasts.kt
package original

// Type checks and automatic casts

fun getStringLength1(obj: Any): Int? {

	if (obj is String) {
		return obj.length
	}

	return null
}

fun getStringLength2(obj: Any): Int? {
	if (obj !is String) return null

	return obj.length
}

fun getStringLength3(obj: Any): Int? {
	if (obj is String && obj.length > 0) {
			return obj.length
	}

	return null
}

fun automaticCastOneVariable(arg1: Int?): Int? {
    if (arg1 != null) {
        return arg1 * 2
    } else {
        return null
    }
}